### PR TITLE
DOC-195 Nested sharing

### PR DESCRIPTION
**Pull request by b0rza**

### CODE REVIEW

File: app/models/repository_manager/repo_item.rb [hunk 1]
```diff
@@ -147,7 +147,7 @@ def can_be_shared_without_nesting?
     ancestor_and_descendant_ids << self.ancestor_ids if !self.ancestor_ids.empty?
 
     # If it exist a sharing, it returns true
-    if RepositoryManager::Sharing.where(repo_item_id: ancestor_and_descendant_ids).count > 0
+    if RepositoryManager::Sharing.where(repo_item_id: ancestor_and_descendant_ids.flatten).count > 0
```
Comment (b0rza) [inline app/models/repository_manager/repo_item.rb:150]: Gets rid of an annoying deprecation message when running tests.

Comment (mbareta) [inline app/models/repository_manager/repo_item.rb:150]: :+1:

File: db/migrate/20131018214212_create_repository_manager.rb [hunk 2]
```diff
@@ -1,36 +0,0 @@
-class CreateRepositoryManager < ActiveRecord::Migration
```
Comment (mbareta) [inline db/migrate/20131018214212_create_repository_manager.rb:1]: Why is this deleted?

**Comment by mbareta** [comment]
:+1:

File: db/migrate/20131018214212_create_repository_manager.rb (hunk 2 above)
Comment (b0rza) [inline db/migrate/20131018214212_create_repository_manager.rb:1]: This is a duplicate migration.
